enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+// +union
+type AWSIngressSpec struct {
+...
+    // +unionDiscriminator
+    // +kubebuilder:validation:Enum:=NLB;Classic
+    // +kubebuilder:validation:Required
+    Type AWSLBType `json:"type,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`
+}
+
+type EIPAllocations string
+
+```
+
+- The API field for `eip-allocations` field in the Ingress Controller CR needs to be set as follows
+  in the file `operator/v1/types_ingress.go`:
+```go
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`        
+}
+
+type EIPAllocations string
+
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+Is the change relevant for standalone clusters?
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory.
+
+This proposal does not affect MicroShift. The proposal
+adds configuration options through API resources and the installer however, it is not
+required for any of those behaviors to be exposed to MicroShift admins through the
+configuration file for MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### 1. Set EIP through installer for the default ingress controller:
+   
+   a. The admin sets the `eip-allocations` in the installer using the install-config.yaml.
+   
+   b. The installer then sets the ingress cluster object with the `eip-allocations`.
+   
+   c. The cluster ingress operator then creates a default ingress controller by:
+
+     i.   creating an default ingress controller CR
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eip-allocations`.
+     
+     iii. then creating a service with service type load balancer with annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+          which uses the value from the field `eip-allocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CRD:
+   
+   The admin sets the `eip-allocations` in the Ingress Controller CR. Then same process
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
```

Comment:
No. We need to recreate the LB service i.e create and recreate. May be we can ask this in the `Open Questions` whether we shall be adding an update option or we can put a note in the docs saying this can cause outage for sometime.

Reply:
https://github.com/openshift/enhancements/pull/1593#discussion_r1544991664

Reply:
@frobware we need to recreate the LB service.